File: priorauth/models/policy.py
from pydantic import BaseModel, Field


class RuleCriterion(BaseModel):
    id: str
    type: str
    description: str
    required: bool = True
    allowed_icd10: list[str] = Field(default_factory=list)


class PolicyRuleDetail(BaseModel):
    id: str
    procedure_cpt: str
    description: str
    category: str
    requires_prior_auth: bool
    criteria: list[RuleCriterion]
    typical_turnaround_days: int | None = None
    auto_approve_if_all_met: bool | None = None
    requires_medical_director_review: bool | None = None


class PolicyOverview(BaseModel):
    payer: str
    plan_type: str
    ruleset_version: str
    last_updated: str
    rules: list[PolicyRuleDetail]
    demo_matching_note: str

File: frontend/src/types/policy.ts
export interface RuleCriterion {
  id: string
  type: string
  description: string
  required: boolean
  allowed_icd10: string[]
}

export interface PolicyRuleDetail {
  id: string
  procedure_cpt: string
  description: string
  category: string
  requires_prior_auth: boolean
  criteria: RuleCriterion[]
  typical_turnaround_days: number | null
  auto_approve_if_all_met: boolean | null
  requires_medical_director_review: boolean | null
}

export interface PolicyOverview {
  payer: string
  plan_type: string
  ruleset_version: string
  last_updated: string
  rules: PolicyRuleDetail[]
  demo_matching_note: string
}

export type DemoScenario = {
  id: string
  title: string
  subtitle: string
  expectedRuleId: string
  expectedOutcome: string
  note: string
}
